From 25c5644164e3286dc722d59c8d7876b1c49c1385 Mon Sep 17 00:00:00 2001
From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>
Date: Thu, 20 Aug 2020 15:14:19 -0400
Subject: [PATCH 3/6] machine_types/numa: compatibility for
 auto_enable_numa_with_memdev

RH-Author: Dr. David Alan Gilbert <dgilbert@redhat.com>
Message-id: <20200820151419.14723-3-dgilbert@redhat.com>
Patchwork-id: 98196
O-Subject: [RHEL-AV 8.3.0 qemu-kvm PATCH 2/2] machine_types/numa: compatibility for auto_enable_numa_with_memdev
Bugzilla: 1849707
RH-Acked-by: Eduardo Habkost <ehabkost@redhat.com>
RH-Acked-by: Igor Mammedov <imammedo@redhat.com>
RH-Acked-by: David Hildenbrand <david@redhat.com>

From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>

The auto_enable_numa_with_memdev flag automatically creates NUMA a
NUMA node in a case like:

    -m 8G,maxmem=16G

but we need it to keep old machine types the same.
This is (mostly) done for upstream machine types in 195784a0cfad.

Power seems to have auto_enable_numa permenantly on anyway.

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 hw/arm/virt.c     | 2 ++
 hw/i386/pc_piix.c | 1 +
 hw/i386/pc_q35.c  | 1 +
 3 files changed, 4 insertions(+)

diff --git a/hw/arm/virt.c b/hw/arm/virt.c
index 26a7920081..26102f22ff 100644
--- a/hw/arm/virt.c
+++ b/hw/arm/virt.c
@@ -2757,6 +2757,7 @@ static void rhel_machine_class_init(ObjectClass *oc, void *data)
     hc->unplug = virt_machine_device_unplug_cb;
     mc->nvdimm_supported = true;
     mc->auto_enable_numa_with_memhp = true;
+    mc->auto_enable_numa_with_memdev = true;
     mc->default_ram_id = "mach-virt.ram";
 
     object_class_property_add(oc, "acpi", "OnOffAuto",
@@ -2860,5 +2861,6 @@ static void rhel820_virt_options(MachineClass *mc)
     compat_props_add(mc->compat_props, hw_compat_rhel_8_2,
                      hw_compat_rhel_8_2_len);
     mc->numa_mem_supported = true;
+    mc->auto_enable_numa_with_memdev = false;
 }
 DEFINE_RHEL_MACHINE(8, 2, 0)
diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index bda2d9ffc8..2415c5edd6 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -1010,6 +1010,7 @@ static void pc_machine_rhel7_options(MachineClass *m)
     m->default_display = "std";
     m->no_parallel = 1;
     m->numa_mem_supported = true;
+    m->auto_enable_numa_with_memdev = false;
     machine_class_allow_dynamic_sysbus_dev(m, TYPE_RAMFB_DEVICE);
     compat_props_add(m->compat_props, pc_rhel_compat, pc_rhel_compat_len);
     m->alias = "pc";
diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index d1e3a9b575..87a0572ec1 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -618,6 +618,7 @@ static void pc_q35_machine_rhel820_options(MachineClass *m)
     m->desc = "RHEL-8.2.0 PC (Q35 + ICH9, 2009)";
     m->alias = NULL;
     m->numa_mem_supported = true;
+    m->auto_enable_numa_with_memdev = false;
     pcmc->smbios_stream_product = "RHEL-AV";
     pcmc->smbios_stream_version = "8.2.0";
     compat_props_add(m->compat_props, hw_compat_rhel_8_2,
-- 
2.27.0

